src/matrix/matrixtdist.jl

```diff
@@ -0,0 +1,196 @@
+"""
+```julia
+MatrixTDist(ν, M, Σ, Ω)
+
+ν:  positive Real
+M:  n x p matrix
+Σ:  n x n positive definite matrix
+Ω:  p x p positive definite matrix
+```
+The [Matrix *t*-Distribution](https://en.wikipedia.org/wiki/Matrix_t-distribution)
+generalizes the Multivariate *t*-Distribution from vectors to matrices. An
+``n\\times p`` matrix ``\\mathbf{X}`` with the matrix *t*-distribution has density
+
+```math
+f(\\mathbf{X} ; \\nu,\\mathbf{M},\\boldsymbol{\\Sigma}, \\boldsymbol{\\Omega}) =
+c_0 \\left|\\mathbf{I}_n + \\boldsymbol{\\Sigma}^{-1}(\\mathbf{X} - \\mathbf{M})\\boldsymbol{\\Omega}^{-1}(\\mathbf{X}-\\mathbf{M})'\\right|^{-\\frac{\\nu+n+p-1}{2}},
+```
+
+where
+
+```math
+c_0=\\frac{\\Gamma_p\\left(\\frac{\\nu+n+p-1}{2}\\right)}{(\\pi)^\\frac{np}{2} \\Gamma_p\\left(\\frac{\\nu+p-1}{2}\\right)} |\\boldsymbol{\\Omega}|^{-\\frac{n}{2}} |\\boldsymbol{\\Sigma}|^{-\\frac{p}{2}}.
+```
+The matrix *t*-distribution arises as the marginal distribution of ``\\mathbf{X}``
+from ``p(\\mathbf{S},\\mathbf{X})=p(\\mathbf{S})p(\\mathbf{X}|\\mathbf{S})``, where
+
+```math
+\\begin{align*}
+\\mathbf{S}&\\sim IW(\\nu + n - 1, \\boldsymbol{\\Sigma})\\\\
+\\mathbf{X}|\\mathbf{S}&\\sim MN(\\mathbf{M}, \\mathbf{S}, \\boldsymbol{\\Omega})
+\\end{align*}
+```
+"""
+struct MatrixTDist{T <: Real, TM <: AbstractMatrix, ST <: AbstractPDMat} <: ContinuousMatrixDistribution
+
+    ν::T
+    M::TM
+    Σ::ST
+    Ω::ST
+
+    logc0::T
+
+end
+
+#  -----------------------------------------------------------------------------
+#  Constructors
+#  -----------------------------------------------------------------------------
+
+function MatrixTDist(ν::T, M::AbstractMatrix{T}, Σ::AbstractPDMat{T}, Ω::AbstractPDMat{T}) where T <: Real
+
+    n, p = size(M)
+
+    ν > 0       || throw(ArgumentError("degrees of freedom must be positive."))
+    n == dim(Σ) || throw(ArgumentError("Number of rows of M must equal dim of Σ."))
+    p == dim(Ω) || throw(ArgumentError("Number of columns of M must equal dim of Ω."))
+
+    logc0 = _matrixtdist_logc0(Σ, Ω, ν)
+    R = Base.promote_eltype(T, logc0)
+    prom_M = convert(AbstractArray{R}, M)
+    prom_Σ = convert(AbstractArray{R}, Σ)
+    prom_Ω = convert(AbstractArray{R}, Ω)
+
+    MatrixTDist{R, typeof(prom_M), typeof(prom_Σ)}(R(ν), prom_M, prom_Σ, prom_Ω, R(logc0))
+
+end
+
+function MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractPDMat, Ω::AbstractPDMat)
+
+    T = Base.promote_eltype(ν, M, Σ, Ω)
+
+    MatrixTDist(convert(T, ν), convert(AbstractArray{T}, M), convert(AbstractArray{T}, Σ), convert(AbstractArray{T}, Ω))
+
+end
+
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractMatrix,         Ω::AbstractMatrix)         = MatrixTDist(ν, M, PDMat(Σ), PDMat(Ω))
```

Comment:
No need to align the argument on every constructor, this tends to make too big diffs every time we modify functions.

Also, some of these constructors can be merged using `Union` type bounds. Example below:
```julia
MatrixTDist(ν::Real, M::AbstractMatrix, Σ::Union{AbstractMatrix, LinearAlgebra.Cholesky}, Ω::Union{AbstractMatrix, LinearAlgebra.Cholesky}) = MatrixTDist(ν, M, PDMat(Σ), PDMat(Ω))
```

Reply:
Awesome. Thanks!